zerver/lib/rest.py

```diff
@@ -46,6 +47,22 @@ def rest_dispatch(request, **kwargs):
             supported_methods[arg] = kwargs[arg]
             del kwargs[arg]
 
+    # Flags can be attached to specific methods by passing a tupule instead of a string to the
+    # methods map in rest_dispatch based url pattern.
+    #
+    # The tuple will be unpacked as a str,list with the string representing the view finction
+    # and the list representing the list of flags.
+    flags = {} # type: Dict[str, List[Any]]
+    for method in list(supported_methods):
+        entry = supported_methods[method]
+
+        if isinstance(entry, tuple):
+            view, view_flags = entry
+            supported_methods[arg] = view
+            for flag in view_flags:
+                if flag in FLAGS:
+                    flags[method] = flag
+
```

Comment:
Can't we just move this inside the `if method_to_use in supported_methods` check, and then in there just do something like:

entry = supported_methods[method_to_use]
if isinstance(entry, tuple):
    (target_function, view_flags) = entry
else:
    target_funtion = entry
    flags = []
...